### ISSUE TITLE
New "Super Jump" tool idea.

### ISSUE THREAD

**Issue by Kagutrenchi** [Zet0rz/nzombies#419]
So, I had an idea about a tool. In some games, the zombies have this type of barrier that allows them to start on one place really low off the map and then jump into the map. You can see this happen with Der Eisendrache in some of the spawns, especially near one of the Wunderfizz spawns. So I thought, why not have that type of thing? It could be this area where, when a zombie walks over it, a zombie would jump to another specified area.

**Comment by DjNathann** [comment]
The zombies will be blocked cause there is a barrier of the "Super jump".

So it would not be a good idea...

**Comment by Kagutrenchi** [comment]
I don't think that you have to have a barrier. Maybe it could be a standalone tool lmb start, rmb finish.

**Comment by Zet0rz** [comment]
> why not have that type of thing?

Because this is extremely difficult to do properly. A custom tool is pretty much necessary, but for that to work you'd need some sort of frame-by-frame teleporting (very resource intensive) to make them follow the path. The alternative (and this is what CoD does) would be to have animations look like they are jumping up and then move them there at the end of the animation. But this would not only need one animation for _every single height you want it to work with_, but also doesn't work with gmods hitbox and navigation system. 

I wish this was possible. I've thought a lot about alternatives, and the best I can come up with for now is just teleporting with some fancy effect.

**Comment by Kagutrenchi** [comment]
I see.

**Comment by Roachisserie** [comment]
Regarding Zet's comment (I haven't got a clue how to quote posts on mobile)

I was planning something similar for R-Base where the bot PlaySequenceAndWait's for a ''preparing to jump'' animation. Then they jump and play a ''gliding'' animation.

The height that they jump is determined by the entity, so while it's entirely possible to make this tool, it'll require a fairly large amount of experimentation to get working right in-game.

**Comment by Zet0rz** [comment]
(You can quote by putting a **>**-symbol at the start)

> I was planning something similar for R-Base where the bot PlaySequenceAndWait's for a ''preparing to jump'' animation. Then they jump and play a ''gliding'' animation.

That's how I'd do it too, but it could be difficult to make the bots know they can navigate that way and also SetPos'ing constantly to glide could look bad and/or be resource intensive. Let me know if you figure something out though, a tool like this would definitely be awesome.

**Comment by Roachisserie** [comment]
SetPos'ing? Oh no I currently use loco.Jump() with the entity feeding the bot the integer on how high to jump

**Comment by Roachisserie** [comment]
I meant gliding as in the HL2 anims name ''glide_all''

**Comment by Zet0rz** [comment]
http://wiki.garrysmod.com/page/CLuaLocomotion/JumpAcrossGap

This is a thing apparently. I can probably experiment with this. Although loco.Jump() itself wouldn't really work, would it? That just causes a straight and mostly laggy jump that is very hard to direct?

**Comment by func-octo** [comment]
> Zetorz said:
But this would not only *need one animation for every single height you want it to work with*,

Actually you could have them place the zombie spawn point and then define a second point, so you could have the spawn point move itself to a position where the animation would show them landing perfectly where the second point was defined, and then teleport the zombie to that second point. I agree it wouldn't cooperate with gMod's hitbox system, at least until they land, which might be fair anyway.
